File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.cpp
#include <string>
#include <vector>


using namespace std;

int solution(int n) {
    int answer = 1;
    
    int half = n / 2 + 2;
    
    
    for(int i = 1; i <= half; i++){
        
        int sum = i;
        for(int j = i+1; j <= half; j++){
            sum += j;
            
            if(sum == n){
                
                answer++;
                break;
            }
            
            if(sum > n){
                
                break;
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.js

function solution(n) {
    
     let count = 1;
    let start = 1, end = 1;
    let sum = 1;
    
    while (start <= n / 2) {
        if (sum < n) {
            end++;
            sum += end;
        } else if (sum > n) {
            sum -= start;
            start++;
        } else {
            count++;
            sum -= start;
            start++;
        }
    }
    
    return count;
}

